core/src/main/scala/org/ensime/core/Project.scala

```diff
@@ -74,7 +74,23 @@ class Project(
   }
   context.actorOf(Props(new ClassfileWatcher(searchService :: reTypecheck :: Nil)), "classFileWatcher")
 
-  def receive: Receive = awaitingConnectionInfoReq
+  // If the system property `ensime.server.async` is missing or not set to 
+  // `true` the old behaviour will be used.
+  private val handleAsyncConnections: Boolean = scala.util.Try(System.getProperty("ensime.server.async") == "true").toOption.getOrElse(false)
```

Comment:
can you please add this to the EnsimeServerConfig with default values in application.conf?

Reply:
also, please invert the name. It should be something like `ensime.legacy.connectionInfoReq = true`